components/CartPopupModal/CartPopupModal.js

```diff
@@ -51,15 +91,29 @@ const CartPopupModal = ({ type, item, selector, setCartState, disabled }) => {
     setSelectedExpiryId(preselectedExpiryId);
   }, [item, hasExpiry, selector]);
 
-  const handleClick = (event) => {
-    setAnchorEl(event.currentTarget);
-  };
-
-  const handleCloseSelector = () => {
-    setAnchorEl(null);
-  };
+  useEffect(() => {
+    if (hasExpiry && type == CART_ITEM_TYPE_DEPOSIT) {
+      if (selectedDate === '') {
+        setItemExpiryDates([
+          ...item.expirydates,
+          { id: 'addNew', expirydate: 'New' },
+        ]);
+      } else {
+        setItemExpiryDates([
+          ...item.expirydates,
+          { id: 'newDate', expirydate: selectedDate },
+          { id: 'addNew', expirydate: 'New' },
+        ]);
+      }
+    } else {
+      setItemExpiryDates(item.expirydates);
+    }
+  }, [selectedDate]);
```

Comment:
try heeding the warning and adding the dependencies if it doesn't lead to an infinite render